src/types.ts

```diff
@@ -216,6 +216,10 @@ function string<Options extends StringOptions>(options?: Options): GetType<strin
 export default {
   /**
    * Creates an array consisting of items of a single type.
+   * Note that the "required" option will not be respected on the type passed to
+   * `array`.
+   * All inner types are "required", i.e. non-optional, by default.
+   * e.g. types.array(types.number({ required: true })) is equivalent to types.array(types.number())
```

Comment:
Let's move this code in an `@example` block below, where you can mention some words with `// comments`